pkg/minikube/machine/start.go

```diff
@@ -288,3 +289,18 @@ func showHostInfo(cfg config.ClusterConfig) {
 	}
 	out.T(out.StartingVM, "Creating {{.driver_name}} {{.machine_type}} (CPUs={{.number_of_cpus}}, Memory={{.memory_size}}MB, Disk={{.disk_size}}MB) ...", out.V{"driver_name": cfg.Driver, "number_of_cpus": cfg.CPUs, "memory_size": cfg.Memory, "disk_size": cfg.DiskSize, "machine_type": machineType})
 }
+
+// AddHostAlias makes fine adjustments to pod resources that aren't possible via kubeadm config.
+func AddHostAlias(c command.Runner, name string, ip net.IP) error {
+	glog.Infof("checking")
+	record := fmt.Sprintf("%s\t%s", ip, name)
+	if _, err := c.RunCmd(exec.Command("grep", record+"$", "/etc/hosts")); err == nil {
+		return nil
+	}
+
+	script := fmt.Sprintf(`{ grep -v '\t%s$' /etc/hosts; echo "%s"; } > /tmp/h.$$; sudo mv /tmp/h.$$ /etc/hosts`, name, record)
```

Comment:
this bit of bash is a little dense, can you add a comment about what it's doing? 

it looks like it's replacing an existing record in /etc/hosts with a new minikube specific one, but I'm not 100% sure

Reply:
agree with @sharifelgamal